### ISSUE TITLE
[Bug] AKShareLoader._fetch_one missing forex path: forex symbols (e.g. EURUSD) fall through to A-share endpoint and fail

### ISSUE THREAD

**Issue by yuzawaya** [HKUDS/Vibe-Trading#54]
## Bug Description

`AKShareLoader._fetch_one` has no forex routing path. Forex symbols such as `EURUSD`, `EURUSD.FX`, or any currency pair not matching `.SZ`/`.SH`/`.BJ`/`.HK`/`.US` fall through to `_fetch_a_share()`, which calls `ak.stock_zh_a_hist()` and fails for forex codes.

The class declares `markets = {"a_share", "us_equity", "hk_equity", "futures", "fund", "macro", "forex"}` — including `"forex"` — but `_fetch_one` has no branch to handle it.

**Note:** This shares the same root cause as #50 (ETF fallback), but affects forex symbols specifically. Reporting separately so forex support is tracked independently.

## Affected Area

- `agent/backtest/loaders/akshare_loader.py` — `_fetch_one` method

## Steps to Reproduce

1. Set up a backtest with `config.json`:
   ```json
   {
     "codes": ["EURUSD"],
     "source": "akshare",
     "start_date": "2023-01-01",
     "end_date": "2025-12-31"
   }
   ```
2. Run the backtest.
3. Observe that AKShare loader silently fails for `EURUSD`.

Minimal reproduction:

```python
# In the vibe-trading agent environment
from backtest.loaders.akshare_loader import DataLoader

loader = DataLoader()
result = loader.fetch(["EURUSD"], "2023-01-01", "2025-12-31")
print(result)  # {} — empty dict, EURUSD silently dropped
```

## Actual Behavior

`fetch()` returns an empty dict `{}` for `EURUSD`. The logger emits:
```
WARNING akshare failed for EURUSD: ...
```
because `_fetch_a_share()` calls `ak.stock_zh_a_hist(symbol="EURUSD")` which is not a valid A-share code.

## Expected Behavior

`AKShareLoader` should route forex symbols (e.g. `EURUSD`, `GBPUSD`, `USDJPY`) to `ak.forex_hist_em(symbol="EURUSD")`, which is available in AKShare and returns valid OHLCV data.

**Verified locally:**

```python
import akshare as ak
df = ak.forex_hist_em(symbol="EURUSD")
print(len(df))   # 7108 rows
print(df.tail(3))
#               日期      代码     名称      今开     最新价      最高      最低    振幅
# 7105  2026-04-23  EURUSD  欧元兑美元  1.1706  1.1684  1.1716  1.1669  0.40
# 7106  2026-04-24  EURUSD  欧元兑美元  1.1686  1.1718  1.1724  1.1673  0.44
# 7107  2026-04-27  EURUSD  欧元兑美元  1.1697  1.1753  1.1756  1.1695  0.52
```

The `symbol_market_map` in `akshare/forex/cons.py` contains `"EURUSD": 119` (190 symbols total, including all major currency pairs).

## Candidate Fix

Add a forex detection function and a `_fetch_forex` method to `AKShareLoader`:

```python
def _is_forex(code: str) -> bool:
    """Detect forex symbols by matching against AKShare's known symbol map."""
    from akshare.forex.cons import symbol_market_map
    return code.upper() in symbol_market_map or code.upper().replace(".FX", "") in symbol_market_map

def _fetch_one(self, code: str, start_date: str, end_date: str, interval: str) -> Optional[pd.DataFrame]:
    import akshare as ak
    if _is_a_share(code):
        return self._fetch_a_share(ak, code, start_date, end_date, interval)
    if _is_us(code):
        return self._fetch_us(ak, code, start_date, end_date)
    if _is_hk(code):
        return self._fetch_hk(ak, code, start_date, end_date)
    if _is_forex(code):           # ← Add this branch
        return self._fetch_forex(ak, code)
    # Default: try A-share
    return self._fetch_a_share(ak, code, start_date, end_date, interval)

def _fetch_forex(self, ak, code: str) -> Optional[pd.DataFrame]:
    """Fetch forex pair via ak.forex_hist_em."""
    symbol = code.upper().replace(".FX", "")  # normalize EURUSD.FX → EURUSD
    try:
        df = ak.forex_hist_em(symbol=symbol)
    except Exception as exc:
        logger.warning("akshare forex_hist_em failed for %s: %s", symbol, exc)
        return None
    if df is None or df.empty:
        return None
    # Normalize: AKShare forex columns are 日期/今开/最新价/最高/最低/振幅
    df = df.rename(columns={"日期": "trade_date", "今开": "open", "最新价": "close", "最高": "high", "最低": "low"})
    df["trade_date"] = pd.to_datetime(df["trade_date"])
    df = df.set_index("trade_date").sort_index()
    df["volume"] = 0.0
    return df[["open", "high", "low", "close", "volume"]]
```

Suggested patch — open to alternative implementations.

## Environment

- **OS:** macOS 25.4 (Darwin 25.4.0)
- **Python:** 3.11.15
- **vibe-trading-ai:** 0.1.5 (installed from PyPI)
- **akshare:** 1.18.57

## Related

- #50: Same root cause (missing branch in `_fetch_one`) — that issue covers ETF symbols; this covers forex symbols

**Comment by warren618** [comment]
Hi @yuzawaya 👋 fixed in b4fa8e9, already on `main`! 🎉

Added a `_is_forex` predicate that consults `akshare.forex.cons.symbol_market_map` (the same source you pointed at 🎯), and a `_fetch_forex` method that calls `ak.forex_hist_em(symbol=…)` with the `.FX` suffix stripped. Routing in `_fetch_one` runs `_is_forex` after the equity branches and before the default A-share fallback, so `EURUSD`, `EURUSD.FX`, lowercase variants, etc. all dispatch correctly. ✅

Column mapping handles the AKShare quirks 🪄 — 最新价 → `close`, volume synthesised as `0.0` since forex doesn't carry volume — and the result is normalised to the project's standard OHLCV schema. Verified locally with real `ak.forex_hist_em(symbol="EURUSD")` — 262 rows for 2024, full backtest runner produces metrics end-to-end.

You were spot-on about this sharing the root cause with #50 (incomplete dispatch table) 💯 — both are fixed in the same commit, plus a related Tushare-init issue that was breaking the fallback chain.

📋 Tests in `tests/test_akshare_loader.py` cover the predicate + routing (mocked, no network).

Thanks so much for the clean repro and reference patch 🙏
